addons/mail/static/src/models/emoji_category.js

```diff
@@ -1,11 +1,30 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { attr, many } from '@mail/model/model_field';
+import { attr, many, one } from '@mail/model/model_field';
+import { replace } from '@mail/model/model_field_command';
+import { categoryTitleByCategoryName } from '@mail/models_data/emoji_data';
 
 registerModel({
     name: 'EmojiCategory',
     identifyingFields: ['categoryName'],
+    recordMethods: {
+        /**
+         * @returns {string}
+         */
+        _computeCategoryTitle() {
+            if (categoryTitleByCategoryName.has(this.categoryName)) {
+                return categoryTitleByCategoryName.get(this.categoryName);
+            }
+            return this.categoryName;
+        },
+        /**
+         * @returns {FieldCommand}
+         */
+        _computeEmojiRegistry() {
+            return replace(this.messaging.emojiRegistry);
+        },
+    },
     fields: {
         allEmojis: many('Emoji', {
```

Comment:
Should properly define a `sort` for all emojis.
Let's say `codepointsRepresentation` in ASC, from top to bottom.